src/packages/solid/utils/createComponentWithTone.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2023 Comcast Cable Communications Management, LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type { Component } from 'solid-js';
+import type { Tone } from '../types/types.js';
+import objectFromEntries from './objectFromEntries.js';
+
+const getToneFromStyles = <ComponentStyle,>(tone: Tone, styles: ComponentStyle) =>
+  objectFromEntries(
+    Object.entries(styles).map(([element, styleObject]) => [element, styleObject.tones?.[tone]])
+  );
+
+const createComponentWithTone = <Props, ComponentStyle>(
+  Component: Component<Props>,
+  tone: Tone,
+  styles: ComponentStyle
+) => {
+  const tonedStyles = getToneFromStyles<ComponentStyle>(tone, styles);
+
+  const TonedComponent = (props: Props) => <Component {...props} toneStyles={tonedStyles} />;
```

Comment:
When you wrap Component you set style=[props.style, tonedStyle] - then you don't need a new non important toneStyles, it's just a style

Reply:
Oh and here you could still do what we had before with style=[props.style, toneStyles[props.tone] - you don't need to lock it down to a specific brand